metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -130,74 +133,132 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
         ]
 
     def _get_datasets(self, catalog: Catalog) -> Iterable[Identifier]:
-        for namespace in catalog.list_namespaces():
-            yield from catalog.list_tables(namespace)
+        namespaces = catalog.list_namespaces()
+        LOGGER.debug(
+            f"Retrieved {len(namespaces)} namespaces, first 10: {namespaces[:10]}"
+        )
+        tables_count = 0
+        for namespace in namespaces:
+            tables = catalog.list_tables(namespace)
+            tables_count += len(tables)
+            LOGGER.debug(
+                f"Retrieved {len(tables)} tables for namespace: {namespace}, in total retrieved {tables_count}, first 10: {tables[:10]}"
+            )
+            yield from tables
 
     def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
-        try:
-            catalog = self.config.get_catalog()
-        except Exception as e:
-            LOGGER.error("Failed to get catalog", exc_info=True)
-            self.report.report_failure("get-catalog", f"Failed to get catalog: {e}")
-            return
+        thread_local = threading.local()
```

Comment:
Is this needed? Why not just pass catalog to _process_dataset?
```
 def _process_dataset(dataset_path, catalog):
```

Reply:
Catalog object can not be used across threads (at least not for every catalog type).